content/docs/reference/programming-model.md

```diff
@@ -115,30 +115,39 @@ The `args` provided to a resource determine what inputs will be used to initiali
 
 All resource constructors also accept an `options` argument which can provide the following additional resource options controlling how the resource will be managed by Pulumi.
 
+###### `additionalSecretOutputs`
+Provides a list of output properties which should be treated as secrets. This value augments any values that Pulumi detects itself, based on what secret inputs to the resource has. This is typically used to express that for a specific instance of a resource, some of its output properties should be treated as secrets (when they would not normally be).
+
+###### `aliases`
+Provides a list of aliases for a resource or component. When making a breaking change to the name or type of a resource or component, you can add the old name to the list of `aliases` for a resource to ensure that existing resources will be migrated to the new name instead of being deleted and replaced with the new named resource.
+
+###### `customTimeouts`
+Provides a set of custom timeouts for create, update, and delete operations on a resource. These timeouts can be specified as a string like "5m", "40s", or "1d" (5 minutes, 40 seconds, or 1 day, respectively).
```

Comment:
Totally minor - but we may want to use code-style for `create`, `update` and `delete` in this sentence to highlight that these are the keys that can be set - and/or add a final sentence:

> For example, `customTimeouts: { create: "1m" }`.